### PR TITLE
Cheaper `dump_mir` take two

### PR DESCRIPTION
**Pull request by oli-obk**
alternative to #105083

r? @nnethercote

### CODE REVIEW

File: compiler/rustc_middle/src/mir/mod.rs [hunk 1]
```diff
@@ -184,30 +184,7 @@ impl RuntimePhase {
 
 impl Display for MirPhase {
```
Comment (nnethercote) [inline compiler/rustc_middle/src/mir/mod.rs:181]: Do we still need this impl?

File: compiler/rustc_mir_transform/src/pass_manager.rs [hunk 2]
```diff
@@ -160,11 +160,9 @@ pub fn dump_mir_for_pass<'tcx>(
     pass_name: &str,
     is_after: bool,
 ) {
-    let phase_index = body.phase.phase_index();
-
     mir::dump_mir(
         tcx,
-        Some(&format_args!("{:03}-{:03}", phase_index, body.pass_count)),
```
Comment (nnethercote) [inline compiler/rustc_mir_transform/src/pass_manager.rs:173]: This one seems fine...

File: compiler/rustc_mir_transform/src/pass_manager.rs [hunk 3]
```diff
@@ -173,14 +171,5 @@ pub fn dump_mir_for_pass<'tcx>(
 }
 
 pub fn dump_mir_for_phase_change<'tcx>(tcx: TyCtxt<'tcx>, body: &Body<'tcx>) {
-    let phase_index = body.phase.phase_index();
-
-    mir::dump_mir(
-        tcx,
-        Some(&format_args!("{:03}-000", phase_index)),
```
Comment (nnethercote) [inline compiler/rustc_mir_transform/src/pass_manager.rs:186]: ... but this one has changed behaviour, I think? `dump_file_basename` does this:
```rust
format!(".{:03}-{:03}", body.phase.phase_index(), body.pass_count)
```
so the digits after the `-` won't be `000`? Unless `phase_index` is somehow always 0 here?

**Comment by nnethercote** [comment]
Looks good except for the question about the `000` above. I'll give you bonus points if you also rename `dump_mir` as `maybe_dump_mir`, which is a better name IMO for a function that usually doesn't dump MIR.

**Comment by oli-obk** [comment]
@JakobDegen I think we talked about pass_count values in a previous PR. Can you have a look at [c7e94b0](https://github.com/rust-lang/rust/pull/105121/commits/c7e94b0efd79a2373902a898c98932bfe0d8547f)?

**Comment by nnethercote** [comment]
r=me if @JakobDegen is happy with the pass_count.

**Comment by JakobDegen** [comment]
This change lgtm. I actually don't quite like the output of this right now, but that's not a result of changes by this PR. I'll wait for this to merge and then try and make some improvements in a follow-up.

**Pull request merged**